plugin/src/main/java/org/opensearch/ml/helper/ConnectorAccessControlHelper.java

```diff
@@ -109,6 +162,71 @@ public void getConnector(Client client, String connectorId, ActionListener<Conne
         }));
     }
 
+    /**
+     * Gets a connector with the provided clients.
+     * @param sdkClient The SDKClient
+     * @param client The OpenSearch client for thread pool management
+     * @param context The Stored Context.  Executing this method will restore this context.
+     * @param getDataObjectRequest The get request
+     * @param connectorId The connector Id
+     * @param listener the action listener to complete with the GetResponse or Exception
+     */
+    public void getConnector(
+        SdkClient sdkClient,
+        Client client,
+        ThreadContext.StoredContext context,
+        GetDataObjectRequest getDataObjectRequest,
+        String connectorId,
+        ActionListener<Connector> listener
+    ) {
+
+        sdkClient
+            .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
```

Comment:
It turns out we are using the single general_thread_pool for all create/get/delete actions. The general_thread_pool is for some infrequent usages with small pool size which might cause some performance issues in edge cases.

Reply:
Do you have any suggestion what should we use here?

Reply:
I ran into issues on flow framework with too-small threadpools I was repurposing. I plan to create a new threadpool just for these client wrapper calls.

Reply:
Yeah, agree with Dan on having a new pool with a larger size which would be better.

Reply:
```
FixedExecutorBuilder sdkClientThreadPool = new FixedExecutorBuilder(
                settings,
                SDK_CLIENT_THREAD_POOL,
                OpenSearchExecutors.allocatedProcessors(settings) * 4,
                10000,
                ML_THREAD_POOL_PREFIX + SDK_CLIENT_THREAD_POOL,
                false
        );
 ```
 
 I'm planning to add this in my subsequent PR and to use this for all the sdkClient related operations. Does this sound good?

Reply:
Yeah, LGTM, thanks.